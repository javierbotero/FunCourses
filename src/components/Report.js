import React from 'react';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Dashboard from '../css/Dashboard.module.css';
import report from '../css/Report.module.css';
import Circle from './Circle';

const Report = props => {
  const {
    user,
    location,
    url,
    findCourses,
    courses,
    setShowMenu,
  } = props;
  const getMetrics = courses => {
    let students = 0;
    let pendings = 0;
    let favorites = 0;
    courses.forEach(c => {
      students += c.confirmed_students.length;
      pendings += c.pending_students.length;
      favorites += c.favorites.length;
    });
    return {
      students,
      pendings,
      favorites,
    };
  };
  const metrics = getMetrics(findCourses(courses, user.courses));
  const months = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];
  const maxCoursesMonth = (courses, max) => {
    let total = 0;
    courses.forEach(c => {
      if (c.dates) {
        const dates = c.dates.split(' ').slice(0, -1);
        if (dates.length > 0) {
          const date = new Date(dates[dates.length - 2]);
          const today = new Date();
          const month = date.getMonth();
          const year = date.getFullYear();
          const todayMonth = today.getMonth();
          const todayYear = today.getFullYear();
          if (month === todayMonth && year === todayYear) {
            total += 1;
          }
        }
      }
    });
    const result = (total / max) * 100;
    return {
      percentage: result > 100 ? 100 : result,
      total,
    };
  };
  return (
    <div>
      <nav className="navMenu">
        <div>
          <Link to={{
            pathname: location.state ? location.state.from.pathname : `${url}`,
            state: { from: location },
          }}
          >
            &#60;
          </Link>
        </div>
        <div>
          Report
        </div>
        <button className="removeButtonStyles" type="button" onClick={() => setShowMenu(true)}>
          <FontAwesomeIcon icon="ellipsis-v" />
        </button>
      </nav>
      <div className={report.report}>
        <header className={`${Dashboard.header} ${report.header}`}>
          <div>
            {'Hello '}
            {user.username}
            {', these are metrics of your actions'}
          </div>
        </header>
        <main className={report.main}>
          <div className={report.field}>
            <div>
              {`Courses opened in: ${months[new Date().getMonth()]}`}
            </div>
            <div>
              <Circle quantity={maxCoursesMonth(
                findCourses(courses, user.courses),
                5,
              )
                .percentage}
              />
              {maxCoursesMonth(findCourses(courses, user.courses)).total}
            </div>
          </div>
          <div className={report.field}>
            <div>
              {'Total Students: '}
            </div>
            <div>
              {metrics.students}
            </div>
          </div>
          <div className={report.field}>
            <div>
              {'Total students waiting for confirmation: '}
            </div>
            <div>
              {metrics.pendings}
            </div>
          </div>
          <div className={report.field}>
            <div>
              {'Total favorites received: '}
            </div>
            <div>
              {metrics.favorites}
            </div>
          </div>
          <div className={report.field}>
            <div>
              {`Taking courses in: ${months[new Date().getMonth()]}`}
            </div>
            <div>
              <Circle
                quantity={maxCoursesMonth(
                  findCourses(courses, user.courses_as_student),
                  10,
                ).percentage}
              />
              {maxCoursesMonth(findCourses(courses, user.courses_as_student)).total}
            </div>
          </div>
          <div className={report.field}>
            <div>
              {'Total friends: '}
            </div>
            <div>
              {user.requests.length + user.pendings.length}
            </div>
          </div>
          <div className={report.field}>
            <div>
              {'Your comments: '}
            </div>
            <div>
              {user.comments.length}
            </div>
          </div>
          <div className={report.field}>
            <div>
              {'Your favorites: '}
            </div>
            <div>
              {user.favorites.length}
            </div>
          </div>
        </main>
      </div>
    </div>
  );
};

Report.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    username: PropTypes.string.isRequired,
    courses_as_student: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      teacher_id: PropTypes.number.isRequired,
      status: PropTypes.string.isRequired,
      dates: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      created_at: PropTypes.string.isRequired,
      updated_at: PropTypes.string.isRequired,
      teacher: PropTypes.shape({
        id: PropTypes.number,
        username: PropTypes.string,
      }).isRequired,
      favorites: PropTypes.arrayOf(PropTypes.shape({
        course_id: PropTypes.number,
        user_id: PropTypes.number,
      })).isRequired,
      subscriptions: PropTypes.arrayOf(PropTypes.shape({
        course_id: PropTypes.number,
        user_id: PropTypes.number.isRequired,
        confirmed: PropTypes.bool,
      })).isRequired,
      comments: PropTypes.arrayOf(PropTypes.shape({
        user_id: PropTypes.number,
        course_id: PropTypes.number,
        body: PropTypes.string,
      })).isRequired,
      pendings: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number,
        user_id: PropTypes.number,
        course_id: PropTypes.number,
        confirmed: PropTypes.bool,
      })).isRequired,
    })).isRequired,
    pending_courses_as_student: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      teacher_id: PropTypes.number.isRequired,
      dates: PropTypes.string.isRequired,
    })).isRequired,
    comments: PropTypes.arrayOf(PropTypes.shape({
      body: PropTypes.string.isRequired,
      course_id: PropTypes.number.isRequired,
    })).isRequired,
    favorites: PropTypes.arrayOf(PropTypes.shape({
      course_id: PropTypes.number,
    })).isRequired,
    pending_to_accept_friendships: PropTypes.arrayOf(PropTypes.shape({
      receiver_id: PropTypes.number,
      sender_id: PropTypes.number,
    })).isRequired,
    friendship_requests: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      username: PropTypes.string,
      email: PropTypes.string,
    })).isRequired,
    requests: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        status: PropTypes.string,
        email: PropTypes.string,
        username: PropTypes.string,
        courses_as_student: PropTypes.arrayOf({
          id: PropTypes.number,
          title: PropTypes.string,
          content: PropTypes.string,
          favorites: PropTypes.arrayOf(PropTypes.shape({
            id: PropTypes.number,
            user_id: PropTypes.number,
            course_id: PropTypes.number,
          })),
        }),
      }),
    ).isRequired,
    pendings: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        status: PropTypes.string,
        email: PropTypes.string,
        username: PropTypes.string,
        courses_as_student: PropTypes.arrayOf({
          id: PropTypes.number,
          title: PropTypes.string,
          content: PropTypes.string,
          favorites: PropTypes.arrayOf(PropTypes.shape({
            id: PropTypes.number,
            user_id: PropTypes.number,
            course_id: PropTypes.number,
          })),
        }),
      }),
    ).isRequired,
    courses: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      teacher_id: PropTypes.number.isRequired,
      status: PropTypes.string.isRequired,
      dates: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      created_at: PropTypes.string.isRequired,
      updated_at: PropTypes.string.isRequired,
      teacher: PropTypes.shape({
        id: PropTypes.number,
        username: PropTypes.string,
      }).isRequired,
      favorites: PropTypes.arrayOf(PropTypes.shape({
        course_id: PropTypes.number,
        user_id: PropTypes.number,
      })).isRequired,
      subscriptions: PropTypes.arrayOf(PropTypes.shape({
        course_id: PropTypes.number,
        user_id: PropTypes.number.isRequired,
        confirmed: PropTypes.bool,
      })).isRequired,
      comments: PropTypes.arrayOf(PropTypes.shape({
        user_id: PropTypes.number,
        course_id: PropTypes.number,
        body: PropTypes.string,
      })),
      pendings: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number,
        user_id: PropTypes.number,
        course_id: PropTypes.number,
        confirmed: PropTypes.bool,
      })).isRequired,
    })),
  }).isRequired,
  courses: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    teacher_id: PropTypes.number.isRequired,
    status: PropTypes.string.isRequired,
    dates: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    created_at: PropTypes.string.isRequired,
    updated_at: PropTypes.string.isRequired,
    teacher: PropTypes.shape({
      id: PropTypes.number,
      username: PropTypes.string,
    }).isRequired,
    favorites: PropTypes.arrayOf(PropTypes.shape({
      course_id: PropTypes.number,
      user_id: PropTypes.number,
    })).isRequired,
    subscriptions: PropTypes.arrayOf(PropTypes.shape({
      course_id: PropTypes.number,
      user_id: PropTypes.number.isRequired,
      confirmed: PropTypes.bool,
    })).isRequired,
    comments: PropTypes.arrayOf(PropTypes.shape({
      user_id: PropTypes.number,
      course_id: PropTypes.number,
      body: PropTypes.string,
    })).isRequired,
    pendings: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      user_id: PropTypes.number,
      course_id: PropTypes.number,
      confirmed: PropTypes.bool,
    })).isRequired,
  })).isRequired,
  location: ReactRouterPropTypes.location.isRequired,
  url: PropTypes.string.isRequired,
  findCourses: PropTypes.func.isRequired,
  setShowMenu: PropTypes.func.isRequired,
};

export default Report;
