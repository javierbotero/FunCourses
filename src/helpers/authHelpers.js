import {
  useState,
  useContext,
  createContext,
} from 'react';
import store from '../store/store';
import { setUserError, removeUserError } from '../reducers/user';

const authContext = createContext();

const useAuth = () => useContext(authContext);

const useAuthState = () => {
  const [userId, setUserId] = useState(localStorage.getItem('currentUserIdFunCourses'));
  const [userPassword, setUserPassword] = useState(localStorage.getItem('currentUserPasswordFunCourses'));
  const authenticate = (
    handleApiRequest,
    initCreator,
    verb,
    url,
    data,
  ) => {
    handleApiRequest(initCreator, verb, url, data)
      .then(resp => (
        resp.json().then(res => {
          if (res.user) {
            localStorage.setItem('currentUserIdFunCourses', res.user.id);
            localStorage.setItem(
              'currentUserPasswordFunCourses',
              'user' in data ? data.user.password : data.password,
            );
            setUserId(res.user.id);
            setUserPassword('user' in data ? data.user.password : data.password);
            store.dispatch(removeUserError());
          } else if (res.error) {
            store.dispatch(setUserError(res.error));
          } else {
            store.dispatch(setUserError('Something went wrong, try later'));
          }
          return res;
        })
      ))
      .catch(err => {
        store.dispatch(setUserError(`Something went wrong. ${err}`));
      });
  };
  return {
    userId,
    userPassword,
    authenticate,
    setUserId,
    setUserPassword,
  };
};

export {
  authContext,
  useAuthState,
  useAuth,
};
