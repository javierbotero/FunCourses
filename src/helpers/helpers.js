import { Link } from 'react-router-dom';
import avatar from '../assets/images/avatar.jpg';
import fun2 from '../assets/images/fun2.jpg';
import { URL } from '../constants/constants';

const initCreator = obj => ({
  method: obj.verb,
  mode: 'cors',
  cache: 'no-cache',
  credentials: 'same-origin',
  headers: {
    'Content-Type': 'application/json',
  },
  redirect: 'follow',
  referrerPolicy: 'no-referrer',
  body: JSON.stringify(obj.data),
});

const objThunk = (url, verb, data) => {
  const init = initCreator({
    verb,
    data,
  });
  return {
    url,
    init,
  };
};

const tokenPayload = (id, token) => ({
  token_id: id,
  token,
});

const userPayload = (id, password) => ({
  current_user_id: id,
  current_user_password: password,
});

const handleApiRequest = async (initCreator, verb, url, data) => {
  const init = initCreator({ verb, data });
  const response = await fetch(url, init)
    .then(res => res)
    .catch(err => err);
  return response;
};

const commentsToDivsWithCourse = (comments, location, url) => comments.map(c => (
  <div key={c.id} className="comment">
    <div className="username">
      {'In: '}
      <Link to={{
        pathname: `${url}/course/${c.course_id}`,
        state: {
          from: location,
        },
      }}
      >
        {c.course.title}
      </Link>
    </div>
    <div className="comment-body">{c.body}</div>
  </div>
));
const commentsToDivs = (comments, location, url) => comments.map(c => (
  <div key={c.id} className="comment">
    <div className="username">
      <Link to={{
        pathname: `${url}/user/${c.user.id}`,
        state: {
          from: location,
          user: c.user,
        },
      }}
      >
        {c.user.username}
      </Link>
    </div>
    <div className="comment-body">{c.body}</div>
  </div>
));
const usersListToDiv = (users, location, url, urlApi) => users.map(u => (
  <div className="user" key={u.id}>
    <div>
      <Link
        to={{
          pathname: `${url}/user/${u.id}`,
          state: {
            from: location,
            user: u,
          },
        }}
      >
        <div
          style={{ backgroundImage: `url(${u.url_avatar ? `${urlApi}${u.url_avatar.slice(1, u.url_avatar.length)}` : avatar}` }}
          className="avatar"
        />
        <div style={{ padding: '5px 0' }}>
          {u.username}
        </div>
      </Link>
    </div>
  </div>
));
const findCourses = (courses, subcourses) => subcourses.map(s => courses.find(c => c.id === s.id));
const findCoursesFromCoursesId = (courses, relation) => relation
  .map(r => courses.find(c => c.id === r.course_id));
const isPresentInUserId = (arr, id) => arr.find(item => item.user_id === parseInt(id, 10));
const isPresentInId = (arr, id) => arr.find(item => item.id === parseInt(id, 10));
const isFriendshipRequested = (arr, id) => arr.find(f => f.receiver_id === parseInt(id, 10));
const mainUrl = c => {
  if (c.main_image_url) {
    return c.main_image_url ? `${URL}${c.main_image_url.slice(1, c.main_image_url.length)}` : fun2;
  }
  return fun2;
};
const picturesToDivs = (images, urlApi, cssClasses) => images.map(img => (
  <div
    key={img.slice(img.length - 30, img.length)}
    className={cssClasses}
    style={{
      backgroundImage: `url(${urlApi}${img})`,
    }}
  />
));

export {
  initCreator,
  objThunk,
  tokenPayload,
  userPayload,
  handleApiRequest,
  commentsToDivsWithCourse,
  commentsToDivs,
  usersListToDiv,
  isPresentInUserId,
  isPresentInId,
  isFriendshipRequested,
  findCourses,
  findCoursesFromCoursesId,
  mainUrl,
  picturesToDivs,
};
