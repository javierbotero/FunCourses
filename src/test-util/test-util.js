import fetchMock from 'fetch-mock';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { URL } from '../constants/constants';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const mockMatch = {
  params: { id: '5' },
  path: 'some-path',
  url: 'url',
};
const mockCourses = [
  {
    id: 5,
    title: 'Karate class',
    content: 'Nice Karate class',
    teacher_id: 7,
    status: 'Open',
    dates: '2021-03-25T00:00:00.000Z 2021-03-29T00:00:00.000Z ',
    price: 10,
    images_url: [],
    created_at: '2021-03-07T17:56:22.487Z',
    updated_at: '2021-03-15T04:21:24.489Z',
    teacher: {
      id: 7,
      username: 'Adele',
    },
    pendings: [],
    pending_students: [],
    favorites: [],
    subscriptions: [],
    confirmed_students: [],
  },
];
const mockHistory = {
  action: 'PUSH',
  block: jest.fn(),
  createHref: jest.fn(),
  go: jest.fn(),
  listen: jest.fn(),
  push: jest.fn(),
  location: {
    hash: '',
    pathname: '',
    search: '',
  },
  replace: jest.fn(),
};
const mockUser = {
  id: 7,
  status: 'Student',
  username: 'Javier 4',
  url_avatar: 'api.com/users/1',
  courses_as_student: [],
  pending_courses_as_student: [],
  comments: [],
  favorites: [],
  pending_to_accept_friendships: [],
  pending_requested_friendships: [],
  friendship_requests: [],
  requests: [],
  pendings: [],
  courses: [
    {
      id: 5,
      title: 'Karate class',
      content: 'Nice Karate class',
      teacher_id: 7,
      status: 'Open',
      dates: '2021-03-25T00:00:00.000Z 2021-03-29T00:00:00.000Z ',
      price: 10,
      created_at: '2021-03-07T17:56:22.487Z',
      updated_at: '2021-03-15T04:21:24.489Z',
      teacher: {
        id: 7,
        username: 'Adele',
      },
      pendings: [],
      pending_students: [],
      favorites: [],
      subscriptions: [],
      confirmed_students: [],
    },
  ],
};
const mockTeacherCourses = [];
const mockLocation = {
  from: { pathname: 'url' },
  hash: 'some-hash',
  pathname: 'url',
  search: 'default',
  state: {
    user: {
      id: 1,
      username: 'Paco',
    },
    from: {
      pathname: 'url',
    },
  },
};
const mockUrl = 'url';
const mockUsersListToDiv = jest.fn();
const mockCommentsToDivs = jest.fn();
const userPassword = '12345';
const func = jest.fn();
const mockUrlApi = 'url';
const mockToken = 'token';
const mockId = 1;

export {
  func,
  mockStore,
  fetchMock,
  mockMatch,
  mockCourses,
  mockTeacherCourses,
  mockLocation,
  mockUrl,
  mockUsersListToDiv,
  mockCommentsToDivs,
  mockUrlApi,
  mockToken,
  mockId,
  mockUser,
  URL,
  mockHistory,
  userPassword,
};
