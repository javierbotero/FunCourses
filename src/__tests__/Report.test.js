import React from 'react';
import {
  render,
  screen,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import * as util from '../test-util/test-util';
import Report from '../components/Report';
import { findCourses } from '../helpers/helpers';

describe('Tests for Report', () => {
  it('prints Hello Javier 4, these are metrics of your actions', () => {
    render(
      <BrowserRouter>
        <div className="desktopMenu" />
        <Report
          setShowMenu={util.func}
          findCourses={findCourses}
          location={util.mockLocation}
          courses={util.mockCourses}
          url={util.mockUrl}
          user={util.mockUser}
        />
      </BrowserRouter>,
    );
    expect(screen.getByText('Hello Javier 4, these are metrics of your actions')).toBeInTheDocument();
  });
  it('prints Total Students:', () => {
    render(
      <BrowserRouter>
        <div className="desktopMenu" />
        <Report
          setShowMenu={util.func}
          findCourses={findCourses}
          location={util.mockLocation}
          courses={util.mockCourses}
          url={util.mockUrl}
          user={util.mockUser}
        />
      </BrowserRouter>,
    );
    expect(screen.getByText('Total Students:')).toBeInTheDocument();
  });
  it('Prints correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <div className="desktopMenu" />
        <Report
          setShowMenu={util.func}
          findCourses={findCourses}
          location={util.mockLocation}
          courses={util.mockCourses}
          url={util.mockUrl}
          user={util.mockUser}
        />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
