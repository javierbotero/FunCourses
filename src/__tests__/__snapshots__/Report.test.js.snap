// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Report Prints correctly 1`] = `
Array [
  <div
    className="desktopMenu"
  />,
  <div>
    <nav
      className="navMenu"
    >
      <div>
        <a
          href="url"
          onClick={[Function]}
        >
          <i
            className="fa chevron-left"
          />
        </a>
      </div>
      <div>
        Report
      </div>
      <button
        className="removeButtonStyles"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa ellipsis-v"
        />
      </button>
    </nav>
    <div
      className="report"
    >
      <header
        className="header header"
      >
        <div>
          Hello 
          Javier 4
          , these are metrics of your actions
        </div>
      </header>
      <main
        className="main"
      >
        <div
          className="field"
        >
          <div>
            Courses opened in April
          </div>
          <div>
            <div
              className="book-child records"
            >
              <svg
                style={
                  Object {
                    "height": 80,
                    "width": 80,
                  }
                }
              >
                <defs>
                  <radialGradient
                    cx="50%"
                    cy="50%"
                    id="gradient"
                    r="70%"
                  >
                    <stop
                      offset="60%"
                      stopColor="#4333B3"
                    />
                    <stop
                      offset="110%"
                      stopColor="#E04534"
                    />
                  </radialGradient>
                </defs>
                <circle
                  cx={40}
                  cy={40}
                  r={28}
                  stroke="#e8e8e8"
                  style={
                    Object {
                      "fill": "transparent",
                      "strokeWidth": 6,
                    }
                  }
                />
                <circle
                  cx={40}
                  cy={40}
                  r={28}
                  stroke="url(#gradient)"
                  style={
                    Object {
                      "fill": "transparent",
                      "strokeDasharray": "175.92918860102841 175.92918860102841",
                      "strokeDashoffset": 175.92918860102841,
                      "strokeWidth": 6,
                      "transform": "rotate(-90deg)",
                      "transformOrigin": "center",
                    }
                  }
                />
              </svg>
            </div>
            0
          </div>
        </div>
        <div
          className="field"
        >
          <div>
            Taking courses in April
          </div>
          <div>
            <div
              className="book-child records"
            >
              <svg
                style={
                  Object {
                    "height": 80,
                    "width": 80,
                  }
                }
              >
                <defs>
                  <radialGradient
                    cx="50%"
                    cy="50%"
                    id="gradient"
                    r="70%"
                  >
                    <stop
                      offset="60%"
                      stopColor="#4333B3"
                    />
                    <stop
                      offset="110%"
                      stopColor="#E04534"
                    />
                  </radialGradient>
                </defs>
                <circle
                  cx={40}
                  cy={40}
                  r={28}
                  stroke="#e8e8e8"
                  style={
                    Object {
                      "fill": "transparent",
                      "strokeWidth": 6,
                    }
                  }
                />
                <circle
                  cx={40}
                  cy={40}
                  r={28}
                  stroke="url(#gradient)"
                  style={
                    Object {
                      "fill": "transparent",
                      "strokeDasharray": "175.92918860102841 175.92918860102841",
                      "strokeDashoffset": 175.92918860102841,
                      "strokeWidth": 6,
                      "transform": "rotate(-90deg)",
                      "transformOrigin": "center",
                    }
                  }
                />
              </svg>
            </div>
            0
          </div>
        </div>
        <div
          className="field"
        >
          <div>
            Total Students: 
          </div>
          <div>
            0
          </div>
        </div>
        <div
          className="field"
        >
          <div>
            Total students waiting for confirmation: 
          </div>
          <div>
            0
          </div>
        </div>
        <div
          className="field"
        >
          <div>
            Total favorites received: 
          </div>
          <div>
            0
          </div>
        </div>
        <div
          className="field"
        >
          <div>
            Total friends: 
          </div>
          <div>
            0
          </div>
        </div>
        <div
          className="field"
        >
          <div>
            Your comments: 
          </div>
          <div>
            0
          </div>
        </div>
        <div
          className="field"
        >
          <div>
            Your favorites: 
          </div>
          <div>
            0
          </div>
        </div>
      </main>
    </div>
  </div>,
]
`;
